Subject: Make strongswan.service a forking service

If the strongswan.service is started with the --nofork option the
systemctl command returns before the service is actually started
successfully. Therefore lets use service type forking to let
systemd wait for the pid file.

Upstream Strongswan has a different way to handle this by offering
a dedicated charon-systemd daemon that has native support for
systemd.

--- a/init/systemd/strongswan.service.in
+++ b/init/systemd/strongswan.service.in
@@ -3,7 +3,9 @@
 After=network-online.target
 
 [Service]
-ExecStart=@SBINDIR@/@IPSEC_SCRIPT@ start --nofork
+Type=forking
+PIDFile=@IPSEC_PIDDIR@/charon.pid
+ExecStart=@SBINDIR@/@IPSEC_SCRIPT@ start
 ExecReload=@SBINDIR@/@IPSEC_SCRIPT@ reload
 StandardOutput=syslog
 Restart=on-abnormal
--- a/init/systemd/Makefile.am
+++ b/init/systemd/Makefile.am
@@ -9,4 +9,5 @@
 	sed \
 	-e "s:@SBINDIR@:$(sbindir):" \
 	-e "s:@IPSEC_SCRIPT@:$(ipsec_script):" \
+	-e "s:@IPSEC_PIDDIR@:$(piddir):" \
 	$(srcdir)/$@.in > $@
